### DISCUSSION TITLE
How does this work on Wayland without XDG-Portals?

### DISCUSSION THREAD

**Discussion by Bitwolfies** [LizardByte/community#315] [Sunshine] [answered]
I'm really curious as I've been lead to believe screen share only works on Wayland due to compositors implementatng their own support via the portals, as Wayland itself lacks it. It also only allows screenshare that's approved via portal Gui, how's it working here?

**Comment by ReenigneArcher** [comment]
I would guess that part of the code was probably borrowed from OBS or some other project. Before my time though so I don't have the background to back it up with any facts.

**Comment by Bitwolfies** [comment]
Obs uses portals though.

**Comment by Bitwolfies** [comment, marked as answer]
Alright, did that whole research thing. Uses KMS which bypasses wayland entirely, and is why the program requires elevated permissions to work. Would love to see it removed in favour of the portals, but its amazing it works at all.

**Reply by ReenigneArcher** [reply]
We would be happy to onboard any devs interested in the Linux side of things. It hasn't gotten much attention lately.

**Reply by Bitwolfies** [reply]
Not complaining really, this blows steam link out of the water, it doesn't work on wayland at all, so the fact anything works is fantastic.
